src/kirin/dialects/ilist/rewrite/unroll.py

```diff
@@ -129,3 +130,10 @@ def rewrite_ForEach(self, node: ForEach) -> RewriteResult:
 
         node.delete()
         return RewriteResult(has_done_something=True)
+
+    def rewrite_Len(self, node: Len):
```

Comment:
why does this needs to live inside unroll rule? it has nothing to do about ilist unroll?

Reply:
I can move it but I will steal that function to get the length of from the type attribute

Reply:
Arguably though this is required for loop unrolling so it kind of makes sense

Reply:
I've move it to `py.len` dialect.